lib/plugins/aws/package/compile/imports/imports.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const traverse = require('traverse');
+
+class AwsCompileImports {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.compileImports = this.compileImports.bind(this);
+
+    this.hooks = {
+      'package:compileImports': this.compileImports,
+    };
+  }
+
+  compileImports() {
+    const that = this; // save this because of the upcoming scope switch
+    traverse(this.serverless.service).forEach(function (propertyParam) {
```

Comment:
We traverse here since the usage for an import (via the variable system) can be everywhere in the `serverless.yml` file.

You can see an example in the service I posted for the verification (https://github.com/serverless/serverless/pull/3475#issue-221802547).